apps/prairielearn/src/pages/instructorQuestions/instructorQuestions.ts

```diff
@@ -73,6 +77,36 @@ router.get(
       return;
     }
 
+    // Special case: if the user accesses the route `.../questions/?qid=...`, we
+    // will redirect them to the preview page for that question.
+    //
+    // This is meant to support automated testing of questions when one might not
+    // want to jump through hoops to get a question ID from a QID.
+    if (req.query.qid && typeof req.query.qid === 'string') {
+      // Find the question they're after.
+      const question = await selectOptionalQuestionByQid({
+        qid: req.query.qid,
+        course_id: res.locals.course.id,
+      });
+
+      if (!question) {
+        throw new error.HttpStatusError(404, 'Question not found');
+      }
+
+      // Forward all query parameters except `qid`. Specifically, we want to support
+      // `variant_seed` for previewing questions with a specific seed.
+      const searchParams = getSearchParams(req);
+      searchParams.delete('qid');
```

Comment:
Armando just learned about the `variant_seed` query param, I figured it'd be nice if we made that Just Work.